src/BalanceClient.ts

```diff
@@ -0,0 +1,24 @@
+import {providers} from "near-api-js";
+
+export class BalanceClient {
+
+  static async balanceOfOnNear(contractId: string, accountId: string): Promise<number> {
+    const {providers} = require("near-api-js");
+    const provider = new providers.JsonRpcProvider(
+      "https://rpc.testnet.near.org"
+    );
+
+    const balanceArgument = "{\"account_id\": \"" + accountId + "\"}";
+    const endodedArgument = Buffer.from(balanceArgument).toString("base64");
+
+    const rawResult = await provider.query({
+      request_type: "call_function",
+      account_id: contractId,
+      method_name: "ft_balance_of",
+      args_base64: endodedArgument,
+      finality: "optimistic",
```

Comment:
final finality?

Reply:
I will check this `finality` field before merging this, this was just assembled to work

Reply:
This is changed to `final` for each contract call